packages/block-library/src/post-comment-content/edit.js

```diff
@@ -1,24 +1,32 @@
 /**
  * WordPress dependencies
  */
+import { __ } from '@wordpress/i18n';
+import { RawHTML } from '@wordpress/element';
 import { useEntityProp } from '@wordpress/core-data';
-import { useBlockProps } from '@wordpress/block-editor';
-
-// TODO: JSDOC types
-export default function Edit( { attributes, context } ) {
-	const { className } = attributes;
-	const { commentId } = context;
+import { useBlockProps, Warning } from '@wordpress/block-editor';
+import { Disabled } from '@wordpress/components';
 
+export default function Edit( { context: { commentId } } ) {
+	const blockProps = useBlockProps();
 	const [ content ] = useEntityProp(
 		'root',
 		'comment',
 		'content',
 		commentId
 	);
-
+	if ( ! content?.rendered ) {
+		return (
+			<div { ...blockProps }>
+				<Warning>{ __( 'Comment has no content.' ) }</Warning>
+			</div>
+		);
+	}
 	return (
-		<div { ...useBlockProps() }>
-			<p className={ className }>{ content }</p>
```

Comment:
Did this used to work somehow before? 🤔 

[The API](https://github.com/WordPress/WordPress/blob/master/wp-includes/rest-api/endpoints/class-wp-rest-comments-controller.php#L1088) seems to return an object from the start?

Reply:
Hi @ntsekouras, I just saw that we are working on the same issue, although it seems that you have advanced more than me :smiley: :sweat_smile: https://github.com/WordPress/gutenberg/pull/35183

> Did this used to work somehow before? :thinking:

Wondering the same ― for me, it wasn't working. Which totally makes sense, as the object the `useEntityProp` hook returns comes from the REST API.

Reply:
Hey David 👋 - sorry for this... I was reviewing the post avatar PR and noticed that it was broken.

I had seen that there was activity to the issue and my PR just fixes the breakage without the rest enhancements in the issue - that's why I just mentioned it as `related`..

I guess we can land this PR and you handle all the other stuff (including the [renaming](https://github.com/WordPress/gutenberg/pull/35183#issuecomment-929222898) 😄 ). What do you think?

Reply:
Yeah, no problem! I'll wait for this PR to be merged. :+1: